### ISSUE TITLE
Fields cannot have name like aBc in contracts due to Jackson

### ISSUE THREAD

**Issue by krissrex** [http4k/http4k#1233]
Another issue from using Jackson for schemas (#1036).
The issue is how jackson uses `USE_STD_BEAN_NAMING` ([javadoc](https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-databind/2.18.2/com/fasterxml/jackson/databind/MapperFeature.html#USE_STD_BEAN_NAMING)) combined with how kotlin names generated getters.

Creating a field in a body, like `rKg` in 
```kotlin
data class MyData(val rKg: Int, val pKg: Int, val gKg: Int)
```
will cause the following error:
```
org.http4k.contract.jsonschema.v3.NoFieldFound: Could not find rkg in MyData(rKg=126000, pKg=97000, gKg=77000)
```

Kotlin will generate `getRKg()`, and Jackson will either name it `rkg` or `RKg`.

---

Workaround:
specify a `renamingStrategy` in `org.http4k.contract.jsonschema.v3.AutoJsonToJsonSchema` in `ApiRenderer.Auto` in `OpenApi3`.

We're using `Kotlinx.Serialization` for the body lens, but it doesn't work for OpenApi rendering, so we have to use Jackson (or others) there.

_Digression:_ A serializer using kotlinx.serialization for the schema as well would be best, but it requires some work in this library. Since the `Api` data class is not `@Serializable`, it can't really be achieved outside the library.

**Comment by daviddenton** [comment]
Yes - Jackson is far too "clever" with this type of thing - it's very annoying for sure.

If you want to PR the renaming strategy (I presume you meant FieldRetrieval?) then we can look at putting that into the stack as standard).

RE: the digression. Eventually (not yet) there will be an overdue rewrite of the contract module where we should be able to more easily render the APIs using Moshi or whatever. This should make it easier to remove jackson as a the only option.
